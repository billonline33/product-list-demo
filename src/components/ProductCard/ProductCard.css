/* listing content */
.listing_content {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.listing_content__element {
  background-color: #ffffff;
  border: 1px solid #e0dfdf;
  margin: 10px;
  width: calc(100% / 4 - 15px);
}
.listing_content__element:nth-child(4n + 1) {
  margin-left: 0px;
}
.listing_content__element:nth-child(4n + 4) {
  margin-right: 0px;
}
.listing_content__element-image {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 15px;
  height: 400px;
}
.listing_content__element-image img {
  max-height: 100%;
  max-width: 100%;
}
.listing_content__element-description {
  margin: 15px;
  text-align: left;
}
.listing_content__element-description h2 {
  color: #6d6b6b;
  font-size: 18px;
  text-align: left;
}
.listing_content__element-description p {
  color: #a3a3a3;
  font-size: 15px;
  padding-top: 4px;
  margin-top: 0px;
  padding-bottom: 9px;
  text-align: left;
}
.listing_content__element-description span {
  color: rgba(43, 43, 43, 0.85);
  font-size: 16px;
  font-weight: bold;
  position: relative;
  bottom: 0;
}

/* Large desktops and laptops */
@media (min-width: 1200px) and (max-width: 1440px) {
  /* listing content */
  .listing_content__element-description h2 {
    font-size: 18px;
  }
  .listing_content__element-description p {
    font-size: 16px;
  }
  .listing_content__element-description span {
    font-size: 16px;
  }
}

/* Portrait tablets and medium desktops */
@media (min-width: 992px) and (max-width: 1199px) {
  /* listing content */
  .listing_content__element {
    margin: 10px;
    width: calc(100% / 4 - 15px);
  }
  .listing_content__element-image {
    padding: 10px;
  }
  .listing_content__element-description {
    margin: 10px;
  }
  .listing_content__element-description h2 {
    font-size: 18px;
  }
  .listing_content__element-description p {
    font-size: 14px;
  }
  .listing_content__element-description span {
    font-size: 16px;
  }
}

/* Portrait tablets and small desktops */
@media (min-width: 768px) and (max-width: 991px) {
  /* listing content */
  .listing_content__element {
    margin: 0px;
    width: calc(100% / 2 - 20px);
  }
  .listing_content__element:nth-of-type(odd) {
    margin: 10px;
  }
  .listing_content__element:nth-of-type(even) {
    margin: 10px;
  }
  .listing_content__element-image {
    padding: 10px;
  }
  .listing_content__element-description {
    margin: 10px;
  }
  .listing_content__element-description h2 {
    font-size: 18px;
  }
  .listing_content__element-description p {
    font-size: 14px;
  }
  .listing_content__element-description span {
    font-size: 16px;
  }
}

/* Landscape phones and portrait tablets */
@media (max-width: 767px) {
  /* listing content */
  .listing_content__element {
    margin: 0px;
    width: calc(100% / 1 - 20px);
  }
  .listing_content__element:nth-of-type(odd) {
    margin: 10px;
  }
  .listing_content__element:nth-of-type(even) {
    margin: 10px;
  }
  .listing_content__element-image {
    padding: 5px;
  }
  .listing_content__element-description {
    margin: 10px;
  }
  .listing_content__element-description h2 {
    font-size: 17px;
  }
  .listing_content__element-description p {
    font-size: 13px;
  }
  .listing_content__element-description span {
    font-size: 15px;
  }
}

/* Landscape phones and smaller */
@media (max-width: 480px) {
  /* listing content */
  .listing_content__element {
    margin: 0px;
    width: calc(100% / 1);
  }
  .listing_content__element:nth-of-type(odd) {
    margin: 5px 0px;
  }
  .listing_content__element:nth-of-type(even) {
    margin: 5px 0px;
  }
  .listing_content__element-image {
    padding: 5px;
  }
  .listing_content__element-description {
    margin: 10px;
  }
  .listing_content__element-description h2 {
    font-size: 16px;
  }
  .listing_content__element-description p {
    font-size: 12px;
  }
  .listing_content__element-description span {
    font-size: 14px;
  }
}
