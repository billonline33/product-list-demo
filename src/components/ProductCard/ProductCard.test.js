import React from 'react';
import ProductCard from './ProductCard';
import { shallow } from 'enzyme';

describe('ProductCard component', () => {
  it('ProductCard renders without crashing', () => {
    expect(shallow(<ProductCard />)).toMatchSnapshot();
  });
});
