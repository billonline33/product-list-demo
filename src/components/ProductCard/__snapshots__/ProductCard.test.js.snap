// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component ProductCard renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="listing_content__element-image"
>
          <img
                    alt="product"
                    src={undefined}
                    width="280px"
          />
</div>,
        <hr />,
        <div
          className="listing_content__element-description"
>
          <h2 />
          <p />
          <span />
</div>,
      ],
      "className": "listing_content__element",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="product"
            src={undefined}
            width="280px"
/>,
          "className": "listing_content__element-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "product",
            "src": undefined,
            "width": "280px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <p />,
            <span />,
          ],
          "className": "listing_content__element-description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="listing_content__element-image"
>
            <img
                        alt="product"
                        src={undefined}
                        width="280px"
            />
</div>,
          <hr />,
          <div
            className="listing_content__element-description"
>
            <h2 />
            <p />
            <span />
</div>,
        ],
        "className": "listing_content__element",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="product"
              src={undefined}
              width="280px"
/>,
            "className": "listing_content__element-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "product",
              "src": undefined,
              "width": "280px",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <p />,
              <span />,
            ],
            "className": "listing_content__element-description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
